bytes/pyproject.toml

```diff
@@ -41,6 +41,7 @@ pyjwt = "^2.8.0"
 fastapi-slim = "^0.111.0"
 structlog = "^24.2.0"
 cryptography = "^43.0.1"
+boto3 = "^1.35.15"
```

Comment:
We could consider creating a separate poetry group for this to make it an optional dependency, so we don't always ship with `boto3` when it's not needed:

```toml
boto3 = { version = "^1.35.15", optional = true }
other_s3_requirement = { version = "^1.2", optional = true }

[tool.poetry.extras]
s3 = ["boto3", "other_s3_requirement"]
```
This can then be installed by `poetry install --extras s3`, for which we should add a build argument to the Dockerfile.

The downside however is that we would have to distribute separate Docker images as well in a release or require users of this functionality to build Bytes from source. A bit of extra work but makes the default installation leaner.

@underdarknl thoughts?

Reply:
Although I like the idea of having contrib and extra package requirements, it does come with a few consequences. For example, we'd end up with more requirements files, and we'd have to modify build pipelines and scripts (e.g., `poetry-dependencies`) and the argument you mentioned. For now, I don't think it's worth the effort if it's only for this functionality (assuming this S3 storage will eventually be part of OpenKAT and not an optional extension). However it's something we could consider for the (near?) future or maybe implement it differently, e.g. storage and middleware as plugins?

Reply:
for any k8s hosted openkat I'd expect them to run on the s3 storage backend, as that's usually a more performant and cheaper (and managed) solution available to these platforms.